_posts/2021-12-13-hhast-4.139.0.markdown

```diff
@@ -0,0 +1,69 @@
+---
+title: "HHAST v4.139.0, do you want more and smarter linters?"
+layout: post
+author: atry
+category: blog
+---
+
+HHAST 4.139.0 is released! This release fixes some minor issues in the `HHClientLinter` introduced by [v4.135.0](https://github.com/hhvm/hhast/releases/tag/v4.135.0).
+
+HHVM 4.126 and above remain supported.
+
+## About HHAST
+
+HHAST is a toolkit for processing the AST of Hack files. It includes a linting framework, along with many built-in linters, serving as an LSP server working with IDE to report lint errors at real time.
+
+## About `HHClientLinter`
```

Comment:
hhclientlinter itself also feels like an implementation detail - it should be part of 'how to enable', perhaps with a brief explanation like "HHClientLinter is the communication layer between HHAST and the typechecker" (not necessarily those words, just meaning to convey the depth/length I'm thinking of)

Give examples of the actual problems that are detected before mentioning codes

It might be useful to say something like 'similar to the typechecker, these new linters use 4-digit numeric codes; to avoid conflicts, the 5xxx range is reserved for these built-in linters'

Reply:
> hhclientlinter itself also feels like an implementation detail - it should be part of 'how to enable', perhaps with a brief explanation like "HHClientLinter is the communication layer between HHAST and the typechecker" (not necessarily those words, just meaning to convey the depth/length I'm thinking of)

How about "`HHClientLinter` is the special linter as a proxy to run linters from the Hack compiler."



> Give examples of the actual problems that are detected before mentioning codes

Example description added.



> It might be useful to say something like 'similar to the typechecker, these new linters use 4-digit numeric codes; to avoid conflicts, the 5xxx range is reserved for these built-in linters'

Error code description added